drud/ddev
# Move release builds to GitHub actions, sign with new Windows cert

## The Problem/Issue/Bug:

* We want to move the full release process from CircleCI to Github Actions
* The current Windows signing key is expiring; the new one requires signing on a self-hosted runner (due to a USB key that's required in the physical machine)

## How this PR Solves The Problem:

Lots of changes to workflow, some changes to Makefile.

## Manual Testing Instructions:

* Review the build, for example, in https://github.com/rfay/ddev/actions/runs/847395359 and https://github.com/rfay/ddev/actions/runs/847395359
* Review the artifacts in, for example, https://github.com/rfay/ddev/releases/tag/v1.18.0-alpha7
    * Download the signed Windows binaries and review the signing of both the installer and ddev.exe itself
    * Download the signed Mac binaries and check the notarization. See https://github.com/drud/signing_tools#validating-the-signature-on-the-binary and https://github.com/drud/signing_tools#validating-notarization

## Release/Deployment notes:

* I haven't tested the push to either AUR or Chocolatey, as there is no dev path for testing these. So they'll have to be checked carefully on next release, and manual cleanup may be required.


<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3003"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
